src/posix/platform/radio_spinel.cpp

```diff
@@ -488,7 +664,12 @@ void RadioSpinel::HandleWaitingResponse(uint32_t          aCommand,
         }
         else
         {
-            if (aCommand == mExpectedCommand)
+            if (aKey == SPINEL_PROP_THREAD_ACTIVE_DATASET ||
```

Comment:
How about making a special format to indicate the caller wants to decode the payload itself? I find `"."` is proper for this purpose. see:https://github.com/openthread/openthread/blob/master/src/ncp/spinel.h#L144